### ISSUE TITLE
feat: Trivy lacks tfsec option `--exclude-downloaded-modules`

### ISSUE THREAD

**Issue by simar7** [aquasecurity/trivy#4618]
### Discussed in https://github.com/aquasecurity/trivy/discussions/4597

<div type='discussions-op-text'>

<sup>Originally posted by **bodgit** June  8, 2023</sup>
### Description

It has become apparent we should migrate from tfsec to Trivy.

We don't do much in the way of tfsec customisation but one option we do use is `--exclude-downloaded-modules`. The theory being there's little value in being notified about issues in third-party modules that we have no control over, and maintaining long lists of `# tfsec:ignore:...` magic comments could get annoying. This option succinctly restricted any issues to just the local codebase.

I can't see an analogous option in Trivy to achieve the same. I can do `--skip-dirs=.terraform` which stops Trivy scanning the downloaded module source, including any examples, etc., but it still flags issues with any `module {}` blocks and I don't really want to have to add long lists of `# trivy:ignore:{this,that,something}` comments that could require maintenance when the module is updated. We have Dependabot automatically raise PRs to update modules and it will break that workflow if someone has to manually add more magic comments to get the PR to pass the Trivy workflow that we would inevitably have, (we currently have a tfsec-based one).

### Desired Behavior

Equivalent functionality to `tfsec --exclude-downloaded-modules`.

### Actual Behavior

External modules are still scanned.

### Reproduction Steps

```
So in my repo, I have the following:

module "some_security_group" {
  source  = "terraform-aws-modules/security-group/aws"
  version = "x.y.z"

  ...
}
Running trivy config --skip-dirs=.terraform . gives me this error:

2023-06-12T12:01:28.948+0100    INFO    Misconfiguration scanning is enabled
2023-06-12T12:01:40.495+0100    INFO    Detected config files: 8

terraform-aws-modules/security-group/aws/main.tf (terraform)

Tests: 5 (SUCCESSES: 4, FAILURES: 1, EXCEPTIONS: 0)
Failures: 1 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 1)

CRITICAL: Security group rule allows egress to multiple public internet addresses.
════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════
Opening up ports to connect out to the public internet is generally to be avoided. You should restrict access to IP addresses or ranges that are explicitly required where possible.

See https://avd.aquasec.com/misconfig/avd-aws-0104
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 terraform-aws-modules/security-group/aws/main.tf:634-641
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── 
628   resource "aws_security_group_rule" "egress_with_cidr_blocks" {
 ...
 634 ┌   cidr_blocks = compact(split(
 635 │     ",",
 636 │     lookup(
 637 │       var.egress_with_cidr_blocks[count.index],
 638 │       "cidr_blocks",
 639 │       join(",", var.egress_cidr_blocks),
 640 └     ),
 ...
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
So Trivy isn't scanning every .tf file in the module source under .terraform, but still seems to be following the module {} blocks and warning me about a resource outside of my control.

(I think https://github.com/aquasecurity/trivy/issues/4581 is related in that it's hard to work out which module {} block is responsible)

Running tfsec --exclude-downloaded-modules . doesn't flag that module block.
```

### Target

Git Repository

### Scanner

Misconfiguration

### Output Format

None

### Mode

Standalone

### Debug Output

```bash
n/a
```

### Operating System

Ubuntu

### Version

```bash
Version: 0.42.1
Policy Bundle:
  Digest: sha256:030345bff4845bae1c89450da85681bc25c0c40d2144a0d9db5254674d155298
  DownloadedAt: 2023-06-08 16:15:38.336264863 +0000 UTC
```

### Checklist

- [ ] Run `trivy --reset`
- [ ] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)</div>

**Comment by bodgit** [comment]
Just tested 0.44.0 and the new `--tf-exclude-downloaded-modules` option seems to be working. Many thanks!

**Comment by cswilliams** [comment]
I just tried this out and while it does reduce the number of issues reported, it's still flagging issues from downloaded modules in the `.terraform/modules/` directory.  For example, I'm using the [S3 bucket module](https://registry.terraform.io/modules/terraform-aws-modules/s3-bucket/aws/latest) and it's still generating issues like the following: 
```
accounts/log_archive/.terraform/modules/us_west_2.s3_access_logs/examples/complete-legacy/main.tf (terraform)
=============================================================================================================
Tests: 1 (SUCCESSES: 0, FAILURES: 1, EXCEPTIONS: 0)
Failures: 1 (UNKNOWN: 0, LOW: 0, MEDIUM: 1, HIGH: 0, CRITICAL: 0)

MEDIUM: Key does not have rotation enabled.
═════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════
You should configure your KMS keys to auto rotate to maintain security and defend against compromise.

See https://avd.aquasec.com/misconfig/avd-aws-0065
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 accounts/log_archive/.terraform/modules/us_west_2.s3_access_logs/examples/complete-legacy/main.tf:21-24
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
  21 ┌ resource "aws_kms_key" "objects" {
  22 │   description             = "KMS key is used to encrypt bucket objects"
  23 │   deletion_window_in_days = 7
  24 └ }
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
```

For now, I'm excluding these directories using the `--skip-dirs` option

**Comment by cswilliams** [comment]
Also, another issue I wanted to report.  I don't appear to be able to set this option in a `trivy.yaml` file. I was hoping to set it in a yaml file so I can run trivy in github actions with this option enabled via the [official action](https://github.com/aquasecurity/trivy-action).

Here's what I tried in the `trivy.yaml` file:
```yaml
scan:
  tf-exclude-downloaded-modules: true
  skip-dirs: 
    - accounts/management/.terraform
    - accounts/log_archive/.terraform
    - accounts/audit/.terraform
    - accounts/shared_services/.terraform
    - accounts/development/.terraform
    - accounts/production/.terraform
    - accounts/staging/.terraform
```

The `skip-dirs` option appears to work but the `tf-exclude-downloaded-modules` does not.

If anyone knows how to set this option in the trivy GHA action, I would be very appreciative. thank you!

**Comment by cswilliams** [comment]
@bodgit That worked! You're right the action is still on `0.43.1` so I'll have to wait on that, but at least I've got a path forward now.  Thanks for the assistance!
